### PR TITLE
TD-1810

### PR DESCRIPTION
**Pull request by m4manjesh**
Closes [TD-1810](https://sce.myjetbrains.com/youtrack/issue/TD-1810/Detail-card-sub-text-color-change)

## Changes

- Updated `DetailCard` component "Updated..." text color to match Figma.
- Convert `DetailCard, FileCard` and `SearchBar` components to typescript.
- Created a `Common.types.ts` file to keep common types
- Increased the title font weight to match Figma

## UI/UX

@Sowbhagya-ipg Please take a look.

On Devel
MicrosoftTeams-image (11) (1)

On this branch
screenshot-localhost_6006-2023 06 14-09_11_34

## Author checklist before assigning a reviewer

- [x] Reviewed my own code-diff.
- [x] PR assigned to me or an appropriate delegate.
- [x] Relevant labels added to the PR.
- [x] Appropriate tests have been added.
- [x] Lint and test workflows pass.

### CODE REVIEW

**Review by nikitharani — CHANGES_REQUESTED** [review]
I have reviewed and I think you missed a few bits to migrate to ts
- [x] Resolve Comments
- [x] stories.tsx files are not completely migrated to ts you just renamed them to tsx. Please look into components that are already ts for example
- [x] FileCard tests are also not migrated to ts
- [ ] index files should be converted to ts and don't forget to export types.

File: src/Card/DetailCard/DetailCard.types.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+export interface DetaiCardHeaderProps {
+  /**
+   * The content of the buttons stack.
+   */
+  buttonsStack?: React.ReactNode;
+  /**
+   * The labels to be displayed on the card.
+   * labels should be an array of objects with the following properties:
+   */
+  labels?: Array<{
```
[the remark this answers is not in the document]
Comment (m4manjesh) [reply src/Card/DetailCard/DetailCard.types.ts:92]: It is already using types of Label

File: src/Card/DetailCard/DetailCard.types.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    _id: string;
+    color: string;
+    description?: string;
+    name: string;
+  }>;
+  /**
+   * Callback fired when the label is clicked.
+   */
+  onClickLabel?: (label: {
```
[the remark this answers is not in the document]
Comment (m4manjesh) [reply src/Card/DetailCard/DetailCard.types.ts:101]: It is already using types of Label

File: src/Card/DetailCard/DetailCard.types.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+   */
+  width: number;
+}
+
+export interface DetailCardLabelStackProps {
+  /**
+   * The labels to be displayed on the card.
+   * labels should be an array of objects with the following properties:
+   */
+  labels?: Array<{
```
[the remark this answers is not in the document]
Comment (m4manjesh) [reply src/Card/DetailCard/DetailCard.types.ts:126]: Is it duplicated comment?

File: src/Card/DetailCard/DetailCard.tsx [hunk 4]
```diff
@@ -93,18 +96,17 @@ function DetailCardHeader({
...
+  const subtitleRef = useRef<HTMLDivElement>(null);
+  const buttonStackRef = useRef<HTMLDivElement>(null);
 
   // label spacing
   const labelSpacing = 8;
 
   // check if title is overflowing
-  const useTitleWidth = titleRef => {
+  const useTitleWidth = (titleRef: React.RefObject<HTMLDivElement>) => {
     // setButtonStackWidth(buttonStackRef.current.clientWidth);
```
[the remark this answers is not in the document]
Comment (nikitharani) [reply src/Card/DetailCard/DetailCard.tsx:111]: this is not resolved yet

**Review by nikitharani — CHANGES_REQUESTED** [review]
Just a small thing please remove the commented code I think you missed to remove. Also, add the Figma link in the PR description next time. Other than this everything LGTM.

**Review by nikitharani — APPROVED** [review]
LGTM. I'll leave you to merge and release.

**Pull request merged**